src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -24,13 +24,14 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Tagging;
 internal sealed partial class TagSpanIntervalTree<TTag>(
```

Comment:
note: my plan fo rthis type is to get rid of it, inlining its underlying IntervalTree into the tagger directly.  The helpers we have here would then be static helper methods in the tagger as they're all tagger specific helpers (most of which can be improved).